File: NPEG/ApplicationExceptions/InfiniteLoopDetectedException.cs
using System;

namespace NPEG.ApplicationExceptions
{
	public class InfiniteLoopDetectedException : NpegException
	{
		public InfiniteLoopDetectedException()
			: base("Supplied grammar rules caused an infinite loop.")
		{
		}

		public InfiniteLoopDetectedException(String message)
			: base(message)
		{
		}

		public InfiniteLoopDetectedException(String message, Exception innerException)
			: base(message, innerException)
		{
		}
	}
}

File: Languages/npeg_javascript/InfiniteLoopDetectedException.js
if(!RobustHaven) var RobustHaven={};
if(!RobustHaven.Text) RobustHaven.Text={};
if(!RobustHaven.Text.Npeg) RobustHaven.Text.Npeg={};

RobustHaven.Text.Npeg = 
{
	InfiniteLoopDetectedException : function(message)
	{
		if (typeof message == "undefined") 
			message = "Supplied grammar rules caused an infinite loop.";
		ParsingExpressionGrammarException.prototype.constructor.call(this, message);

	}
	InfiniteLoopDetectedException.prototype = new ParsingExpressionGrammarException;
	InfiniteLoopDetectedException.prototype.constructor = InfiniteLoopDetectedException;
}